File: projetmanga-client-android/app/src/main/java/com/mangaproject/data/model/Manga.kt
package com.mangaproject.data.model

import com.google.gson.annotations.SerializedName

data class Manga(
    @SerializedName("_id")
    val id: String,
    val nom: String,
    val description: String?,
    val dateDeSortie: String?,
    val urlImage: String,
    val auteur: String,
    val genres: List<String>?,
    val jikanId: Int? = null,
    val source: String? = null,
    val chapitres: List<String>? = emptyList()
)

data class MangaUpdateRequest(
    val nom: String,
    val description: String?,
    val urlImage: String?,
    val dateDeSortie: String?,
    val auteur: String,
    val genres: List<String>,
    val chapitres: List<String>? = emptyList()
)

data class MangaDetailResponse(
    val message: String?,
    val manga: Manga
)

File: projetmanga-server/models/Manga.js
const mongoose = require('mongoose');

const MangaSchema = new mongoose.Schema({
    nom: { type: String, required: true, trim: true },
    description: String,
    dateDeSortie: Date,
    urlImage: {type: String,default:'https://www.istockphoto.com/photos/placeholder-image'},
    auteur: { type: String, required: true, trim: true },

    genres: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Genre' }],
    chapitres: [{ type: String }] ,// stocke directement les URLs des images

    jikanId: { type: Number, default: null },
    source: { type: String, enum: ["local", "jikan"], default: "local" },

}, { timestamps: true });



module.exports = mongoose.model('Manga', MangaSchema);
